Argo/Types/Decoded/Alternative.swift

```diff
@@ -13,8 +13,16 @@
 infix operator <|> { associativity left precedence 140 }
 
 public func <|> <T>(lhs: Decoded<T>, rhs: Decoded<T>) -> Decoded<T> {
-  if case .Success = lhs {
-    return lhs
+  return lhs.alternative(rhs)
+}
+
+public extension Decoded {
+  func alternative<T>(other: Decoded<T>) -> Decoded<T> {
+    if case .Success = self {
+      return self as! Decoded<T>
+    } else {
+      return other
+    }
   }
-  return rhs
 }
+
```

Comment:
Trailing Newline Violation: Files should have a single trailing newline. (trailing_newline)